tracer/src/Datadog.Trace/AppSec/Waf/Encoder.cs

```diff
@@ -57,30 +57,32 @@ public static ObjType DecodeArgsType(DDWAF_OBJ_TYPE t)
+        private static string TruncateLongString(string s) =>
+            s.Length > MaxStringLength ? s.Substring(0, MaxStringLength) : s;
 
-        public Obj Encode(object o, List<Obj> argCache) => EncodeInternal(o, argCache, MaxObjectDepth);
+        public Obj Encode(object o, List<Obj> argCache, bool applySafetyLimits) =>
+            EncodeInternal(o, argCache, MaxObjectDepth, applySafetyLimits);
 
-        private Obj EncodeInternal(object o, List<Obj> argCache, int remainingDepth)
+        private Obj EncodeInternal(object o, List<Obj> argCache, int remainingDepth, bool applyLimits)
         {
             var value =
                 o switch
                 {
-                    null => CreateNativeString(string.Empty),
-                    string s => CreateNativeString(s),
-                    JValue jv => CreateNativeString(jv.Value?.ToString() ?? string.Empty),
-                    int i => new Obj(_wafNative.ObjectSigned(i)),
-                    long i => new Obj(_wafNative.ObjectSigned(i)),
-                    uint i => new Obj(_wafNative.ObjectUnsigned(i)),
-                    ulong i => new Obj(_wafNative.ObjectUnsigned(i)),
-                    IEnumerable<KeyValuePair<string, JToken>> objDict => EncodeDictionary(objDict.Select(x => new KeyValuePair<string, object>(x.Key, x.Value)), argCache, remainingDepth),
-                    IEnumerable<KeyValuePair<string, string>> objDict => EncodeDictionary(objDict.Select(x => new KeyValuePair<string, object>(x.Key, x.Value)), argCache, remainingDepth),
-                    IEnumerable<KeyValuePair<string, List<string>>> objDict => EncodeDictionary(objDict.Select(x => new KeyValuePair<string, object>(x.Key, x.Value)), argCache, remainingDepth),
-                    IEnumerable<KeyValuePair<string, string[]>> objDict => EncodeDictionary(objDict.Select(x => new KeyValuePair<string, object>(x.Key, x.Value)), argCache, remainingDepth),
-                    IEnumerable<KeyValuePair<string, object>> objDict => EncodeDictionary(objDict, argCache, remainingDepth),
-                    IList<JToken> objs => EncodeList(objs.Select(x => (object)x), argCache, remainingDepth),
-                    IList<string> objs => EncodeList(objs.Select(x => (object)x), argCache, remainingDepth),
-                    IList<object> objs => EncodeList(objs, argCache, remainingDepth),
+                    null => CreateNativeString(string.Empty, applyLimits),
+                    string s => CreateNativeString(s, applyLimits),
+                    JValue jv => CreateNativeString(jv.Value?.ToString() ?? string.Empty, applyLimits),
+                    int i => CreateNativeString(i.ToString(), applyLimits),
+                    long i => CreateNativeString(i.ToString(), applyLimits),
+                    uint i => CreateNativeString(i.ToString(), applyLimits),
+                    ulong i => CreateNativeString(i.ToString(), applyLimits),
```

Comment:
These have changed from `new Obj(_wafNative.ObjectSigned(i)),` to `CreateNativeString(i.ToString(), applyLimits),`. Is that intentional?

Reply:
Yes, we've signed / unsigned shouldn't be used, since the WAF is basically a regex engine, it only cares about string data. I'll remove the underlying functions too.